### PR TITLE
Badge element

### LINKED ISSUE #252: Add badge element component
## Problem

Some use-cases have come up for this type of thing. Both as a label and as a mini button.

https://getbootstrap.com/docs/4.5/components/badge/

## Proposed

A uids take on the badge/pill that can also be used for links and that does not conflict with other frameworks.

### PR DESCRIPTION
**Pull request by joewhitsitt**
Resolves #252 

Was planning to use as part of https://github.com/uiowa/uiowa/issues/1137

### CODE REVIEW

File: src/components/badge/badge.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
+@import '../../assets/scss/variables';
+
+.badge {
+  display: inline-block;
+  padding: 0 7px;
```
Comment (bspeare) [inline src/components/badge/badge.scss:5]: These look good!  I think I would like to see a little extra space on these like:
```
    padding: 0 9px;
    line-height: 20px;
```

**Comment by bspeare** [comment]
Also, should we add a variant of how it looks inline with a heading tag?   It looks like we might need to adjust line-heights. 

`<h2>` might need some adjustments to work.  

Screen Shot 2020-06-24 at 9 29 24 AM

Here is the bootstrap page:

Screen Shot 2020-06-24 at 9 28 00 AM

**Comment by joewhitsitt** [comment]
Screen Shot 2020-06-24 at 10 25 49 AM

With relative sizing i got as close to your pixel recommendations as possible for the default. 

Screen Shot 2020-06-24 at 10 27 15 AM

**Comment by joewhitsitt** [comment]
Sounds good. I will copy and wire up badge in uids_base as part of https://github.com/uiowa/uiowa/pull/1827 for now. Thanks for the flexibility.

**Comment by pyrello** [comment]
@bspeare and @joewhitsitt Sounds good to me.

**Pull request merged**